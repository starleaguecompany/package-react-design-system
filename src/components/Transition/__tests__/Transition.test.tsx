import * as React from 'react'
import { render, screen, waitFor } from '@testing-library/react'

import { Collapse, Fade, ScaleFade, SlideFade } from '../index'

describe('Collapse Transition', () => {
  test('should render correctly with no props', () => {
    render(<Collapse data-qa="Collapse" />)

    expect(screen.getByTestId('Collapse')).toMatchSnapshot()
    expect(screen.getByTestId('Collapse')).not.toBeVisible()
  })

  test('should render correctly with attributes', () => {
    render(<Collapse data-qa="Collapse" id="test-id" className="test-class" />)

    expect(screen.getByTestId('Collapse')).toMatchSnapshot()
    expect(screen.getByTestId('Collapse')).toHaveClass('test-class')
    expect(screen.getByTestId('Collapse')).toHaveAttribute('id', 'test-id')
  })

  test('should render correctly with content', () => {
    render(<Collapse data-qa="Collapse">Test content</Collapse>)

    expect(screen.getByTestId('Collapse')).toMatchSnapshot()
    expect(screen.getByTestId('Collapse')).toHaveTextContent('Test content')
  })

  test('should render correctly with visible prop', () => {
    render(
      <Collapse data-qa="Collapse" visible>
        Test content
      </Collapse>
    )

    expect(screen.getByTestId('Collapse')).toMatchSnapshot()
    expect(screen.getByTestId('Collapse')).toBeVisible()
  })

  test('should render correctly with unmountOnExit prop', () => {
    const { queryByTestId } = render(
      <Collapse data-qa="Collapse" unmountOnExit>
        Test content
      </Collapse>
    )

    expect(queryByTestId('Collapse')).toBeNull()
  })
})

describe('Fade Transition', () => {
  test('should render correctly with no props', () => {
    render(<Fade data-qa="Fade" />)

    expect(screen.getByTestId('Fade')).toMatchSnapshot()
    expect(screen.getByTestId('Fade')).not.toBeVisible()
  })

  test('should render correctly with attributes', () => {
    render(<Fade data-qa="Fade" id="test-id" className="test-class" />)

    expect(screen.getByTestId('Fade')).toMatchSnapshot()
    expect(screen.getByTestId('Fade')).toHaveClass('test-class')
    expect(screen.getByTestId('Fade')).toHaveAttribute('id', 'test-id')
  })

  test('should render correctly with content', () => {
    render(<Fade data-qa="Fade">Test content</Fade>)

    expect(screen.getByTestId('Fade')).toMatchSnapshot()
    expect(screen.getByTestId('Fade')).toHaveTextContent('Test content')
  })

  test('should render correctly with visible prop', async () => {
    render(
      <Fade data-qa="Fade" visible>
        Test content
      </Fade>
    )

    expect(screen.getByTestId('Fade')).toMatchSnapshot()
    await waitFor(() => expect(screen.getByTestId('Fade')).toBeVisible())
  })

  test('should render correctly with unmountOnExit prop', () => {
    const { queryByTestId } = render(
      <Fade data-qa="Fade" unmountOnExit>
        Test content
      </Fade>
    )

    expect(queryByTestId('Fade')).toBeNull()
  })
})

describe('ScaleFade Transition', () => {
  test('should render correctly with no props', () => {
    render(<ScaleFade data-qa="ScaleFade" />)

    expect(screen.getByTestId('ScaleFade')).toMatchSnapshot()
    expect(screen.getByTestId('ScaleFade')).not.toBeVisible()
  })

  test('should render correctly with attributes', () => {
    render(<ScaleFade data-qa="ScaleFade" id="test-id" className="test-class" />)

    expect(screen.getByTestId('ScaleFade')).toMatchSnapshot()
    expect(screen.getByTestId('ScaleFade')).toHaveClass('test-class')
    expect(screen.getByTestId('ScaleFade')).toHaveAttribute('id', 'test-id')
  })

  test('should render correctly with content', () => {
    render(<ScaleFade data-qa="ScaleFade">Test content</ScaleFade>)

    expect(screen.getByTestId('ScaleFade')).toMatchSnapshot()
    expect(screen.getByTestId('ScaleFade')).toHaveTextContent('Test content')
  })

  test('should render correctly with visible prop', async () => {
    render(
      <ScaleFade data-qa="ScaleFade" visible>
        Test content
      </ScaleFade>
    )

    expect(screen.getByTestId('ScaleFade')).toMatchSnapshot()
    await waitFor(() => expect(screen.getByTestId('ScaleFade')).toBeVisible())
  })

  test('should render correctly with unmountOnExit prop', () => {
    const { queryByTestId } = render(
      <ScaleFade data-qa="ScaleFade" unmountOnExit>
        Test content
      </ScaleFade>
    )

    expect(queryByTestId('ScaleFade')).toBeNull()
  })
})

describe('SlideFade Transition', () => {
  test('should render correctly with no props', () => {
    render(<SlideFade data-qa="SlideFade" />)

    expect(screen.getByTestId('SlideFade')).toMatchSnapshot()
    expect(screen.getByTestId('SlideFade')).not.toBeVisible()
  })

  test('should render correctly with attributes', () => {
    render(<SlideFade data-qa="SlideFade" id="test-id" className="test-class" />)

    expect(screen.getByTestId('SlideFade')).toMatchSnapshot()
    expect(screen.getByTestId('SlideFade')).toHaveClass('test-class')
    expect(screen.getByTestId('SlideFade')).toHaveAttribute('id', 'test-id')
  })

  test('should render correctly with content', () => {
    render(<SlideFade data-qa="SlideFade">Test content</SlideFade>)

    expect(screen.getByTestId('SlideFade')).toMatchSnapshot()
    expect(screen.getByTestId('SlideFade')).toHaveTextContent('Test content')
  })

  test('should render correctly with visible prop', async () => {
    render(
      <SlideFade data-qa="SlideFade" visible>
        Test content
      </SlideFade>
    )

    expect(screen.getByTestId('SlideFade')).toMatchSnapshot()
    await waitFor(() => expect(screen.getByTestId('SlideFade')).toBeVisible())
  })

  test('should render correctly with unmountOnExit prop', () => {
    const { queryByTestId } = render(
      <SlideFade data-qa="SlideFade" unmountOnExit>
        Test content
      </SlideFade>
    )

    expect(queryByTestId('SlideFade')).toBeNull()
  })
})
