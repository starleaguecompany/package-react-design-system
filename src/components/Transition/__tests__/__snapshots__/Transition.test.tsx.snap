// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse Transition should render correctly with attributes 1`] = `
<div
  class="test-class"
  data-qa="Collapse"
  id="test-id"
  style="overflow: hidden; display: block; opacity: 0; height: 0px;"
/>
`;

exports[`Collapse Transition should render correctly with content 1`] = `
<div
  data-qa="Collapse"
  style="overflow: hidden; display: block; opacity: 0; height: 0px;"
>
  Test content
</div>
`;

exports[`Collapse Transition should render correctly with no props 1`] = `
<div
  data-qa="Collapse"
  style="overflow: hidden; display: block; opacity: 0; height: 0px;"
/>
`;

exports[`Collapse Transition should render correctly with visible prop 1`] = `
<div
  data-qa="Collapse"
  style="overflow: hidden; display: block; opacity: 1; height: auto;"
>
  Test content
</div>
`;

exports[`Fade Transition should render correctly with attributes 1`] = `
<div
  class="test-class"
  data-qa="Fade"
  id="test-id"
  style="opacity: 0;"
/>
`;

exports[`Fade Transition should render correctly with content 1`] = `
<div
  data-qa="Fade"
  style="opacity: 0;"
>
  Test content
</div>
`;

exports[`Fade Transition should render correctly with no props 1`] = `
<div
  data-qa="Fade"
  style="opacity: 0;"
/>
`;

exports[`Fade Transition should render correctly with visible prop 1`] = `
<div
  data-qa="Fade"
  style="opacity: 0;"
>
  Test content
</div>
`;

exports[`ScaleFade Transition should render correctly with attributes 1`] = `
<div
  class="test-class"
  data-qa="ScaleFade"
  id="test-id"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
/>
`;

exports[`ScaleFade Transition should render correctly with content 1`] = `
<div
  data-qa="ScaleFade"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  Test content
</div>
`;

exports[`ScaleFade Transition should render correctly with no props 1`] = `
<div
  data-qa="ScaleFade"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
/>
`;

exports[`ScaleFade Transition should render correctly with visible prop 1`] = `
<div
  data-qa="ScaleFade"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
>
  Test content
</div>
`;

exports[`SlideFade Transition should render correctly with attributes 1`] = `
<div
  class="test-class"
  data-qa="SlideFade"
  id="test-id"
  style="opacity: 0; transform: translateX(0px) translateY(8px) translateZ(0);"
/>
`;

exports[`SlideFade Transition should render correctly with content 1`] = `
<div
  data-qa="SlideFade"
  style="opacity: 0; transform: translateX(0px) translateY(8px) translateZ(0);"
>
  Test content
</div>
`;

exports[`SlideFade Transition should render correctly with no props 1`] = `
<div
  data-qa="SlideFade"
  style="opacity: 0; transform: translateX(0px) translateY(8px) translateZ(0);"
/>
`;

exports[`SlideFade Transition should render correctly with visible prop 1`] = `
<div
  data-qa="SlideFade"
  style="opacity: 0; transform: translateX(0px) translateY(8px) translateZ(0);"
>
  Test content
</div>
`;
